gui/packages/desktop/src/main/index.js

```diff
@@ -237,38 +248,39 @@ const ApplicationMain = {
   async _onDaemonConnected() {
     this._connectedToDaemon = true;
 
-    // reset the reconnect backoff when connection established.
-    this._reconnectBackoff.reset();
+    // subscribe to events
+    try {
+      await this._subscribeEvents();
+    } catch (error) {
+      log.error(`Failed to subscribe: ${error.message}`);
 
-    if (this._windowController) {
-      this._windowController.send('daemon-connected');
+      return this._recoverFromBootstrapError(error);
```

Comment:
FYI Calls to register event subscriptions, fetch initial tunnel state and settings must not fail. If that happens, the error recovery mechanism is triggered which basically starts the lifecycle from scratch.